core/services/gateway/message.go

```diff
@@ -0,0 +1,27 @@
+package gateway
+
+import "encoding/json"
+
+/*
+ * Top-level Message structure containing:
+ *   - universal fields identifying the request, the sender and the target DON/service
+ *   - product-specific payload
+ */
+type Message struct {
+	MessageId string          `json:"message_id"`
+	Method    string          `json:"method"`
+	DonId     string          `json:"don_id"`
+	ServiceId string          `json:"service_id"`
+	Sender    string          `json:"sender"`
+	Signature string          `json:"signature"`
+	Payload   json.RawMessage `json:"payload"`
+}
+
+func DecodeMessage(msgBytes []byte) (msg *Message, err error) {
```

Comment:
You could use `json.Marshaler` interface like we do for many other custom types in Core:

e.g.
```
// MarshalJSON implements the json.Marshaler interface.
func (m Message) MarshalJSON() ([]byte, error) {
}

func (m *Message) UnmarshalJSON(data []byte) error {
}
```

Reply:
@pinebit turns out I don't even need that interface if I put my Message type inside the RPC types. Let me know if the current version is OK.

I additionally added a nested "MessageBody" struct, like it was suggested in the CLIP.